src/controllers/crate_owner_invitation.rs

```diff
@@ -1,28 +1,64 @@
 use super::frontend_prelude::*;
 
-use crate::models::{CrateOwner, CrateOwnerInvitation, OwnerKind};
-use crate::schema::{crate_owner_invitations, crate_owners};
-use crate::views::{EncodableCrateOwnerInvitation, InvitationResponse};
+use crate::models::{CrateOwner, CrateOwnerInvitation, OwnerKind, User};
+use crate::schema::{crate_owner_invitations, crate_owners, users};
+use crate::views::{EncodableCrateOwnerInvitation, EncodablePublicUser, InvitationResponse};
+use diesel::dsl::any;
+use std::collections::HashSet;
+use std::iter::FromIterator;
 
 /// Handles the `GET /me/crate_owner_invitations` route.
 pub fn list(req: &mut dyn RequestExt) -> EndpointResult {
-    let user_id = req.authenticate()?.user_id();
-    let conn = &*req.db_read_only()?;
+    // Ensure that the user is authenticated
+    let user = req.authenticate()?.user();
 
+    // Load all pending invitations for the user
+    let conn = &*req.db_read_only()?;
     let crate_owner_invitations: Vec<CrateOwnerInvitation> = crate_owner_invitations::table
-        .filter(crate_owner_invitations::invited_user_id.eq(user_id))
+        .filter(crate_owner_invitations::invited_user_id.eq(user.id))
         .load(&*conn)?;
+
+    // Make a list of all related users and deduplicate it by using a `HashSet`
+    let mut user_ids = HashSet::new();
+    for invitation in &crate_owner_invitations {
+        user_ids.insert(invitation.invited_by_user_id);
+    }
+
+    // Load all related users
+    let users: Vec<User> = users::table
+        .filter(users::id.eq(any(Vec::from_iter(user_ids))))
+        .load(conn)?;
+
+    // Turn `CrateOwnerInvitation` list into `EncodableCrateOwnerInvitation` list
     let crate_owner_invitations = crate_owner_invitations
         .into_iter()
-        .map(|i| EncodableCrateOwnerInvitation::from(i, conn))
+        .map(|invitation| {
+            let inviter_id = invitation.invited_by_user_id;
+            let inviter_name = users
+                .iter()
+                .find(|user| user.id == inviter_id)
+                .map(|user| user.gh_login.clone())
```

Comment:
This could result in bad performance if there are a lot of users and a lot of invitations. Could we convert the `Vec<User>` above into a `HashMap<i32, User>` to get fast lookups?